tests/unit/test_example_data.py

```diff
@@ -0,0 +1,40 @@
+from pathlib import Path
+
+import pytest
+
+from ehrql.file_formats.csv import CSVRowsReader
+from ehrql.query_model.column_specs import get_column_specs_from_schema
+from ehrql.query_model.introspection import get_table_nodes
+from ehrql.tables import core, tpp
+
+
+filenames = [
+    "addresses.csv",
+    "clinical_events.csv",
+    "medications.csv",
+    "ons_deaths.csv",
+    "patients.csv",
+    "practice_registrations.csv",
+]
+
+ql_tables = [
+    tpp.addresses,
+    tpp.clinical_events,
+    tpp.medications,
+    core.ons_deaths,
+    core.patients,
+    tpp.practice_registrations,
+]
+
+
+@pytest.mark.parametrize("filename,ql_table", zip(filenames, ql_tables))
+def test_read_rows(filename, ql_table):
+    table_nodes = get_table_nodes(ql_table._qm_node)
+    [table] = table_nodes  # There should only be one table
+    column_specs = get_column_specs_from_schema(table.schema)
+
+    CSVRowsReader(
+        Path(f"ehrql/example-data/{filename}"),
+        column_specs=column_specs,
+        allow_missing_columns=True,
+    )
```

Comment:
This test ends up duplicating some of the logic from `LocalFileQueryEngine` where it might be better to use that logic directly. It would also avoid the awkwardness of having to manually specific the filenames.

You could do this with something like:
```py
LocalFileQueryEngine("path/to/example-data").populate_database([ql_table._qm_node])
```
Which will throw an error if there's anything wrong with the data for that tabe.